packages/cli/src/templates/expo.ts

```diff
@@ -0,0 +1,277 @@
+import chalk from "chalk";
+import { Listr } from "listr2";
+import { existsSync } from "node:fs";
+import { writeFile } from "node:fs/promises";
+import { TemplateOptions } from "../commands/create";
+import { Template } from "./template";
+
+import { exec } from "node:child_process";
+import { promisify } from "node:util";
+const execAsync = promisify(exec);
+export interface ExpoContext {
+  options: TemplateOptions;
+}
+
+export const Expo: Template = {
+  name: "expo",
+  description: "A new expo typescript project with gluestack-ui renderer",
+  async handler(options) {
+    await new Listr<ExpoContext>([
+      {
+        title: "Create directory",
+        task: async ({ options: { cwd } }, task) => {
+          task.title += ` ${cwd}`;
+          if (existsSync(cwd)) {
+            throw new Error(`Directory ${cwd} already exists`);
+          }
+          await execAsync(
+            `npx create-expo-app ${cwd} -t expo-template-blank-typescript`,
+          );
+        },
+      },
+      {
+        title: "Create Expo project",
+        task: async ({ options: { cwd, name } }) => {
+          await writeFile(`${cwd}/app.json`, APP_JSON_CONTENT(name), "utf8");
+          await writeFile(`${cwd}/App.tsx`, APP_TSX_CONTENT, "utf8");
+          await writeFile(
+            `${cwd}/babel.config.js`,
+            BABEL_CONFIG_CONTENT,
+            "utf8",
+          );
+          await writeFile(
+            `${cwd}/metro.config.js`,
+            METRO_CONFIG_CONTENT,
+            "utf8",
+          );
+          await writeFile(
+            `${cwd}/package.json`,
+            PACKAGE_JSON_CONTENT(name),
```

Comment:
I realize that some of these files will have been created with the `create-expo cli` command, so I can remove the writing of this package.json file, and some of the packages I add below that come with the creation of the expo app.